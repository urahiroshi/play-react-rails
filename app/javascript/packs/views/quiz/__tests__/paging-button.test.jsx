/* eslint react/prop-types: 0 */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PagingButton from '../paging-button'

const createComponent = ({ disabled }) => (
  <PagingButton
    className="quiz-prev"
    disabled={disabled}
    onClick={() => {}}
  >
    <span className="glyphicon glyphicon-chevron-left" />
  </PagingButton>
)

describe('#render', () => {
  it('should show snapshot', () => {
    const wrapper = shallow(createComponent({ disabled: false }))
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  describe('when disabled === false', () => {
    it('not have disabled class', () => {
      const wrapper = shallow(createComponent({ disabled: false }))
      expect(wrapper.find('.disabled').exists()).toBe(false)
    })
  })

  describe('when disabled === true', () => {
    it('have disabled class when disabled === true', () => {
      const wrapper = shallow(createComponent({ disabled: true }))
      expect(wrapper.find('.disabled').exists()).toBe(true)
    })
  })
})
