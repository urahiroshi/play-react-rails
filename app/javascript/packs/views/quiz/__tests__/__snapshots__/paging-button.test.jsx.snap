// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render should show snapshot 1`] = `
<div
  className="quiz-prev"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    className="glyphicon glyphicon-chevron-left"
  />
</div>
`;
